lib/css-tools.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+var csstree     = require('css-tree'),
+    stable      = require('stable'),
+    specificity = require('csso/lib/restructure/prepare/specificity');
+
+
+function flattenToSelectors(cssAst) {
+  var selectors   = [],
+      curSelector = null;
+
+  // TODO: walkup to avoid look-behind
+  csstree.walk(cssAst, function(node, item, list) {
+
+    // selector with pseudos
+    // look-behind: The pseudo selectors come after the affected simple selector
+    if(node.type == 'PseudoClassSelector' || 
+       node.type == 'PseudoElementSelector') {
+      curSelector.pseudos.push({ item: item, list: list });
+    }
+
+    // (simple) selector
+    if(node.type !== 'Selector') {
+      return;
+    }
+
+    curSelector = {
+      item:    item,
+      atrule:  this.atrule,
+      rule:    this.rule,
+      pseudos: []
+    };
+    selectors.push(curSelector);
+
+  });
+
+  return selectors;
+}
+
+
+function filterByMqs(selectors, useMqs) {
+  return selectors.filter(function(selector) {
+    if(selector.atrule === null) {
+      return useMqs.indexOf('') > -1;
+    }
+
+    var mqName = selector.atrule.name;
+    var mqExpr = csstree.translate(selector.atrule.expression);
```

Comment:
It makes sense only when `selector.atrule.expression`'s type is `MediaQueryList`